components/select_resource_type.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import {gql, useQuery} from "@apollo/client";
+
+
+const SelectResourceType: any = (props: { handleSelect: Function }) => {
+
+    const queryTypes = gql`
+            query {
+              resourceSpecifications{
+                name
+                id
+              }
+            }
+          `
+
+    const types: Array<{ id: string, name: string }> = useQuery(queryTypes).data?.resourceSpecifications
+
+    const options = () => types?.map((type) => (<option key={type.id} value={type.id}>
```

Comment:
althought this is almost readable... arrow functions of arrow functions are not easy to read... `=> ... =>` should be avoided